### PR TITLE
Feature/unr 1376 add handover disable option

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Add runtime option to disable handover.

#### Release note
Added runtime option to disable property handover when running in non-zoned deployments.

#### Tests
Ran against the unified project.

#### Primary reviewers
@improbable-valentyn @aleximprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 1]
```diff
@@ -161,6 +163,11 @@ void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
 
 	ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 	{
+		if (bEnableHandover == false && Type == SCHEMA_Handover)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp]:
```suggestion
		if (!bEnableHandover && Type == SCHEMA_Handover)
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 2]
```diff
@@ -191,6 +198,11 @@ void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
 
 		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 		{
+			if (bEnableHandover == false && Type == SCHEMA_Handover)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp]:
```suggestion
			if (!bEnableHandover && Type == SCHEMA_Handover)
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 3]
```diff
@@ -344,7 +344,7 @@ bool FSpatialGDKEditorToolbarModule::ValidateGeneratedLaunchConfig() const
 	{
 		if (SpatialGDKRuntimeSettings->bUsingQBI && (*EnableChunkInterest == TEXT("true")))
 		{
-			const EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(FString::Printf(TEXT("The legacy flag \"enable_chunk_interest\" is set to true in the generated launch configuration. This flag needs to be set to false when QBI is enabled.\n\nDo you want to configure your launch config settings now?"))));
+			const EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(TEXT("The legacy flag \"enable_chunk_interest\" is set to true in the generated launch configuration. This flag needs to be set to false when QBI is enabled.\n\nDo you want to configure your launch config settings now?")));
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Nice

Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: We should get these messages reviewed by tech writers.

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: I'd like to get this in tonight, if there's no other objections. I'll create a follow up ticket.

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: That's totally fine, I didn't mean that this should block the merge.

**Comment by improbable-valentyn** [comment]
So just to clarify, this will stop generating handover components when generating schema, which should be sufficient for disabling handover at runtime?

**Comment by m-samiec** [comment]
> So just to clarify, this will stop generating handover components when generating schema, which should be sufficient for disabling handover (from happening at runtime)?

No, all schema are still generated, their ids are stripped out from the classinfo at runtime though, so it's as if they don't exist.

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 4]
```diff
@@ -19,11 +19,13 @@ USpatialGDKSettings::USpatialGDKSettings(const FObjectInitializer& ObjectInitial
 	, ActorReplicationRateLimit(0)
 	, EntityCreationRateLimit(0)
 	, OpsUpdateRate(1000.0f)
+	, bEnableHandover(true)
 	, bUsingQBI(true)
 	, PositionUpdateFrequency(1.0f)
 	, PositionDistanceThreshold(100.0f) // 1m (100cm)
 	, bEnableMetrics(true)
 	, MetricsReportRate(2.0f)
+	, bUseFrameTimeAsLoad(false)
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]: Looks like it's not supposed to be in this PR? I can't find any place it's used.

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 5]
```diff
@@ -59,6 +59,10 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
 	UPROPERTY(EditAnywhere, config, Category = "Replication", meta = (ConfigRestartRequired = false, DisplayName = "SpatialOS Network Update Rate"))
 	float OpsUpdateRate;
 
+	/** Replicate handover properties between servers, required for zoned worker deployments.*/
+	UPROPERTY(EditAnywhere, config, Category = "Replication", meta = (ConfigRestartRequired = false))
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: Should we add DisplayName for this property?

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: `bEnableHandover` becomes `Enable Handover` with the default conversion, which should be good enough?

Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: What does ConfigRestartRequired mean? Is it true that it's not required for any of the options?

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 6]
```diff
@@ -377,6 +377,20 @@ bool FSpatialGDKEditorToolbarModule::ValidateGeneratedLaunchConfig() const
 		return false;
 	}
 
+	int32 PlayNumberOfServers;
+	GetDefault<ULevelEditorPlaySettings>()->GetPlayNumberOfServers(PlayNumberOfServers);
+	if (!SpatialGDKRuntimeSettings->bEnableHandover && PlayNumberOfServers > 1)
+	{
+		const EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(TEXT("Property handover is disabled and multiple launch servers are specified.\nThis is not supported.\n\nDo you want to configure your project settings now?")));
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: As I understand - the value of handover option affects schema generation. Should it be mentioned here (and maybe somewhere else?) that when changing the option - schema must be regenerated?

**Comment by improbable-valentyn** [comment]
> > So just to clarify, this will stop generating handover components when generating schema, which should be sufficient for disabling handover (from happening at runtime)?
> 
> No, all schema are still generated, their ids are stripped out from the classinfo at runtime though, so it's as if they don't exist.

That's great. I like this even better now.

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp (hunk 4 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]: This is a driveby - it wasn't defaulted previously.

**Pull request merged**